config/config.go

```diff
@@ -136,6 +114,39 @@ func Parse(reader io.Reader, location string) (conf Config, err error) {
 	return conf, err
 }
 
+// replaceEnvVars replaces environment variable placeholders in reader stream with values
+// i.e. "val = $VAR" -> "val = 3"
+func replaceEnvVars(reader io.Reader) (io.Reader, error) {
+	// Variable definition follows IEEE Std 1003.1-2001
+	//   A dollar sign ($) followed by an upper-case letter, followed by
+	//   zero or more upper-case letters, digits, or underscores (_).
+	regexStr := `\$[A-Z]+[A-Z1-9_]*`
```

Comment:
@gdey brought up another approach, using [os.Expand](https://golang.org/pkg/os/#Expand) which would be cross platform and cleaner than the regex approach.

Reply:
Yeah, looking at that now. I would say go with the regex, but enhance the regex to deal with `${SHELL_NAME}` as well. Or We require that all shell var substitution is done with `${}` to make it more obvious. In which case we would still want the regex.  (For reference: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html )